### ISSUE TITLE
building //tools/... crashes

### ISSUE THREAD

**Issue by janakdr** [bazelbuild/bazel#940]
I synced with master after a long time, then ran ./compile.sh test. Then I copied output/bazel to ~/bin and ran ~/bin/bazel test -k -- //... -//src/java_tools/singlejar:zipTests

I got a huge number of errors like:

ERROR: /Users/janakr/bazel/out/com/google/devtools/build/lib/BUILD:423:1: Target '//out/com/google/devtools/build/lib:util' contains an error and its package is in error and referenced by '//out/com/google/devtools/build/lib:bazel-core'.

(I think the original errors were about packages not existing.) Then there was a crash:

java.lang.IllegalArgumentException: Multiple entries with same key: tools/build_rules/go/toolchain=/Users/janakr/bazel and tools/build_rules/go/toolchain=/Users/janakr/bazel
	at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:136)
	at com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:98)
	at com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:84)
	at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:295)
	at com.google.devtools.build.lib.buildtool.BuildTool.transformPackageRoots(BuildTool.java:302)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:210)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:335)
	at com.google.devtools.build.lib.runtime.commands.TestCommand.doTest(TestCommand.java:119)
	at com.google.devtools.build.lib.runtime.commands.TestCommand.exec(TestCommand.java:104)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:371)
	at com.google.devtools.build.lib.runtime.BlazeRuntime$3.exec(BlazeRuntime.java:1021)
	at com.google.devtools.build.lib.server.RPCService.executeRequest(RPCService.java:65)
	at com.google.devtools.build.lib.server.RPCServer.executeRequest(RPCServer.java:434)
	at com.google.devtools.build.lib.server.RPCServer.serve(RPCServer.java:229)
	at com.google.devtools.build.lib.runtime.BlazeRuntime.serverMain(BlazeRuntime.java:976)
	at com.google.devtools.build.lib.runtime.BlazeRuntime.main(BlazeRuntime.java:773)
	at com.google.devtools.build.lib.bazel.BazelMain.main(BazelMain.java:56)

**Comment by janakdr** [comment]
Seems to be totally reproducible by running bazel build -k -- //tools/...

**Comment by damienmg** [comment]
Repro?

**Comment by janakdr** [comment]
Is what I wrote not a repro? :)

**Comment by damienmg** [comment]
Your top package is out and I don't know how you got there so yes a more precise repro would be welcomed :)

**Comment by janakdr** [comment]
I'm running from the bazel/ root directory. Not sure what you mean by "top
package is out". I think the out/ error messages in my initial report were
unrelated, since as I said in my second message, it crashes with just bazel
build -k //tools/...

**Comment by damienmg** [comment]
On your Google workstation?

**Comment by janakdr** [comment]
macbook air.

**Comment by damienmg** [comment]
This has freezed my shell on my mac book air. We definitely have issues
there.

**Comment by damienmg** [comment]
Ok I was able to repro after a restart.

**Comment by damienmg** [comment]
Assigning to Ulf, it seems related to the work for interleaving loading and analysis phase

**Comment by bsilver8192** [comment]
This part looks like what I'm fixing in [I8bd7f7b00bec58a7157507595421bc50c81b404c](https://bazel-review.googlesource.com/#/q/I8bd7f7b00bec58a7157507595421bc50c81b404c) due to loading a single package with both `@//package:target` and `//package:target`:
```
java.lang.IllegalArgumentException: Multiple entries with same key: tools/build_rules/go/toolchain=/Users/janakr/bazel and tools/build_rules/go/toolchain=/Users/janakr/bazel
at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:136)
at com.google.common.collect.RegularImmutableMap.checkNoConflictInKeyBucket(RegularImmutableMap.java:98)
at com.google.common.collect.RegularImmutableMap.fromEntryArray(RegularImmutableMap.java:84)
at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:295)
at com.google.devtools.build.lib.buildtool.BuildTool.transformPackageRoots(BuildTool.java:302)
at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:210)
at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:335)
at com.google.devtools.build.lib.runtime.commands.TestCommand.doTest(TestCommand.java:119)
at com.google.devtools.build.lib.runtime.commands.TestCommand.exec(TestCommand.java:104)
at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:371)
at com.google.devtools.build.lib.runtime.BlazeRuntime$3.exec(BlazeRuntime.java:1021)
at com.google.devtools.build.lib.server.RPCService.executeRequest(RPCService.java:65)
at com.google.devtools.build.lib.server.RPCServer.executeRequest(RPCServer.java:434)
at com.google.devtools.build.lib.server.RPCServer.serve(RPCServer.java:229)
at com.google.devtools.build.lib.runtime.BlazeRuntime.serverMain(BlazeRuntime.java:976)
at com.google.devtools.build.lib.runtime.BlazeRuntime.main(BlazeRuntime.java:773)
at com.google.devtools.build.lib.bazel.BazelMain.main(BazelMain.java:56)
```

**Comment by damienmg** [comment]
This was fixed by d7d6d62
